### ISSUE TITLE
HTTP upload performance issues

### ISSUE THREAD

**Issue by Sammers21** [artipie/artipie#382]
I have deployed the [sample project](https://github.com/artipie/artipie/tree/master/examples/maven/sample-for-deployment) into artipie and nexus. Nexus performs ~19 times better.

![image](https://user-images.githubusercontent.com/16746106/88260879-eb85d500-cccd-11ea-8e62-818a8aba3122.png)

**Comment by g4s8** [comment]
@Sammers21 Maven client sends `Expect: 100-continue` header by default and waiting for the response before sending the body for few seconds. We need to respond with `100` status code for this kind of requests if authentication is valid and accept payloads on next request. Maven sends this request before payload to make sure that authentication headers are valid. See https://maven.apache.org/guides/mini/guide-http-settings.html

It can be disabled from `settings.xml` with `server` configuration:
```
    <server>
      <id>artipie</id>
      <configuration>
        <httpConfiguration>
          <put>
            <params>
              <property>
                <name>http.protocol.expect-continue</name>
                <value>%b,false</value>
              </property>
            </params>
          </put>
        </httpConfiguration>
      </configuration>
    </server>
```
After these changes in `settings.xml`, the example `mvn -f ./sample-for-deployment deploy -Dmaven.install.skip=true ` completed in `1.050s`.

@artipie/contributors FYI.

**Comment by Sammers21** [comment]
@g4s8, I wasn't using any options with nexus.

**Comment by g4s8** [comment]
Should be fixed by #402
